app/views/play/cinematic/DialogSystem.js

```diff
@@ -0,0 +1,314 @@
+import anime from 'animejs/lib/anime.es.js'
+import { AnimeCommand, SyncFunction } from './Command/AbstractCommand'
+import { getClearText, getTextPosition, getSpeaker } from '../../../schemas/selectors/cinematic'
+import utils from 'app/core/utils'
+import tmpl from 'tmpl'
+
+const SVGNS = 'http://www.w3.org/2000/svg'
+const padding = 10
+
+/**
+ * This system coordinates drawing HTML and SVG to the screen.
+ * It is also responsible for localization and interpolation of the speech bubbles.
+ */
+export default class DialogSystem {
+  constructor ({ canvasDiv, camera }) {
+    const div = this.div = document.createElement('div')
+    const width = camera.canvasWidth
+    const height = camera.canvasHeight
+    div.style.width = `${width}px`
+    div.style.height = `${height}px`
+
+    const svg = this.svg = document.createElementNS(SVGNS, 'svg')
+    svg.setAttribute('xmlns', SVGNS)
+    svg.setAttribute('height', `${height}`)
+    svg.setAttribute('width', `${width}`)
+
+    svg.style.pointerEvents = div.style.pointerEvents = 'none'
+    svg.style.position = div.style.postition = 'absolute'
+
+    canvasDiv.appendChild(svg)
+    canvasDiv.appendChild(div)
+
+    this.shownDialogBubbles = []
+    this._context = {}
+  }
+
+  /**
+   * This context object can be accessed from the dialog text templates.
+   * E.g. a context object of: `{ name: 'Mary' }` can then be used in the following
+   * dialog text:
+   *   `Hello, <%=o.name%>`
+   * Which then appears as:
+   *   `Hello, Mary`
+   *
+   * @param {Object} context - used for interpolation.
+   */
+  set templateContext (context) {
+    this._context = context
```

Comment:
Why are we not setting this in the constructor? Also is it generally called template context in the language of text interpolation, otherwise the name `context` doesnt look intuitive to me.

Reply:
I like `templateData`, or maybe `templateDataParameters` since parameter is the term used in their documentation.